pyrenew_hew/site_level_dynamics_model.py

```diff
@@ -0,0 +1,566 @@
+        self.p_hosp_w_sd_rv = p_hosp_w_sd_rv
+        self.autoreg_p_hosp_rv = autoreg_p_hosp_rv
+        self.hosp_wday_effect_rv = hosp_wday_effect_rv
+        self.inf_to_hosp_rv = inf_to_hosp_rv
+        self.log10_g_rv = log10_g_rv
+        self.mode_sigma_ww_site_rv = mode_sigma_ww_site_rv
+        self.sd_log_sigma_ww_site_rv = sd_log_sigma_ww_site_rv
+        self.ww_site_mod_sd_rv = ww_site_mod_sd_rv
+        self.phi_rv = phi_rv
+        self.ww_ml_produced_per_day = ww_ml_produced_per_day
+        self.pop_fraction = pop_fraction
+        self.ww_uncensored = ww_uncensored
+        self.ww_censored = ww_censored
+        self.ww_sampled_lab_sites = ww_sampled_lab_sites
+        self.ww_sampled_subpops = ww_sampled_subpops
+        self.ww_sampled_times = ww_sampled_times
+        self.ww_log_lod = ww_log_lod
+        self.lab_site_to_subpop_map = lab_site_to_subpop_map
+        self.hosp_times = hosp_times
+        self.max_ww_sampled_days = max_ww_sampled_days
+        self.include_ww = include_ww
+
+        self.inf_with_feedback_proc = InfectionsWithFeedback(
+            infection_feedback_strength=infection_feedback_strength_rv,
+            infection_feedback_pmf=infection_feedback_pmf_rv,
+        )
+
+        self.ar_diff_rt = DifferencedProcess(
+            fundamental_process=ARProcess(),
+            differencing_order=1,
+        )
+
+        return None
+
+    def validate(self):  # numpydoc ignore=GL08
+        return None
+
+    def sample(
+        self,
+        n_datapoints=None,
+        data_observed_hospital_admissions=None,
+        data_observed_log_conc=None,
+    ):  # numpydoc ignore=GL08
+        if n_datapoints is None:
+            if (
+                data_observed_hospital_admissions is None
+                and data_observed_log_conc is None
+            ):
+                raise ValueError(
+                    "Either n_datapoints or data_observed_hosp_admissions "
+                    "must be passed."
+                )
+            elif (
+                data_observed_hospital_admissions is None
+                and data_observed_log_conc is not None
+            ):
+                raise ValueError(
+                    "Either n_datapoints or data_observed_hosp_admissions "
+                    "must be passed."
+                )
+            elif (
+                data_observed_hospital_admissions is not None
+                and data_observed_log_conc is None
+            ):
+                n_datapoints = len(data_observed_hospital_admissions)
+            else:
+                n_datapoints = max(
+                    len(data_observed_hospital_admissions),
+                    self.max_ww_sampled_days,
+                )
+        else:
+            if (
+                data_observed_hospital_admissions is not None
+                or data_observed_log_conc is not None
+            ):
+                raise ValueError(
+                    "Cannot pass both n_datapoints and "
```

Comment:
To do forecasting, you pass the number of datapoints, rather than the data itself.

Reply:
The total number of data points you want in the calibration + nowcast/forecast period? 

This would still be when you are using the data to calibrate the model, correct? It's just if you want to produce estimates that extend beyond the time range of the data, you would need to tell the model the total number of time points you want to generate estimates for?

Reply:
Right! You pass the number of datapoints when calling predictive methods on the model after it has been calibrated.
Or if you just want to run forward simulations form the model, you pass the number of datapoints.

Reply:
You cannot comforting and forecasting at the same time. You can fit by passing the actual data. You can create a (prior or posterior) predictive distribution by passing the number of time points (which can be arbitrarily few or many).

Reply:
Ok I interpreted your "it just works" statement incorrectly then. 

To simulate from the posterior predictive, do you pass in a fitted model object and then the total number of time points you want to generate posterior predictive observations for (e.g. calibration + nowcast/forecast)? I'm just struggling to see the issue with generating data beyond the number of days of the observed hospital admissions that @sbidari described in the PR description.